e2e/test/FileUploadE2ETests.cs

```diff
@@ -54,6 +59,90 @@ public async Task FileUpload_X509_SmallFile_Http()
             await UploadFile(Client.TransportType.Http1, smallFile, true).ConfigureAwait(false);
         }
 
+        [TestMethod]
+        [TestCategory("LongRunning")]
+        public async Task FileUpload_SmallFile_Http_GranularSteps()
+        {
+            string filename = await GetTestFileNameAsync(FileSizeSmall).ConfigureAwait(false);
+            using FileStream fileStreamSource = new FileStream(filename, FileMode.Open, FileAccess.Read);
+            Http1TransportSettings fileUploadTransportSettings = new Http1TransportSettings();
+
+            await UploadFileGranular(fileStreamSource, filename, fileUploadTransportSettings).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        [TestCategory("LongRunning")]
+        public async Task FileUpload_SmallFile_Http_GranularSteps_x509()
+        {
+            string filename = await GetTestFileNameAsync(FileSizeSmall).ConfigureAwait(false);
+            using FileStream fileStreamSource = new FileStream(filename, FileMode.Open, FileAccess.Read);
+            Http1TransportSettings fileUploadTransportSettings = new Http1TransportSettings();
+
+            await UploadFileGranular(fileStreamSource, filename, fileUploadTransportSettings, true).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        [TestCategory("LongRunning")]
+        public async Task FileUpload_SmallFile_Http_GranularSteps_Proxy()
+        {
```

Comment:
Even if this is a service feature, I would just check if the following behavior works and SDK did not impose any restrictions on it with new implemention.
1. 10 (get this number from what msdn says if it is incorrect) fileuploads in parallel

Reply:
We shouldn't be documenting service side limitations in our SDK. They could change on a whim and without anyone telling us to update

Reply:
This isn't exercising a new path on the service, so the restrictions are still the same

Reply:
I have not asked for documentation in the comment. My comment was to "I would just check if the following behavior works and SDK did not impose any restrictions on it with new implemention."

Reply:
I did check, and everything still works on the same quotas. No new restrictions are brought on from this